### DISCUSSION TITLE
"WARN  GET https://... error (underfined) ..." on my Raspberry Pi 4b

### DISCUSSION THREAD

**Discussion by ideaexplorer233** [pnpm/pnpm#7433] [Q&A] [answered]
## What's happened

I installed pnpm via `sudo pacman -S pnpm` if you need to know.

```
[oceancat@raspi4 ~]$ pnpm create astro@latest
 WARN  GET https://registry.npmjs.org/sisteransi/-/sisteransi-1.0.5.tgz error (undefined). Will retry in 10 seconds. 2 retries left.
 WARN  GET https://registry.npmjs.org/colorette/-/colorette-2.0.20.tgz error (undefined). Will retry in 10 seconds. 2 retries left.

[some similar messages]

 WARN  GET https://registry.npmjs.org/signal-exit/-/signal-exit-3.0.7.tgz error (undefined). Will retry in 10 seconds. 2 retries left.
.../share/pnpm/store/v3/tmp/dlx-683229   |  +39 ++++
 ERROR  Worker pnpm#2 exited with code 1

pnpm: Worker pnpm#2 exited with code 1
    at Worker.<anonymous> (/usr/lib/node_modules/pnpm/dist/pnpm.cjs:81630:27)
    at Worker.emit (node:events:515:28)
    at [kOnExit] (node:internal/worker:315:10)
    at Worker.<computed>.onexit (node:internal/worker:229:20)
 WARN  GET https://registry.npmjs.org/emoji-regex/-/emoji-regex-9.2.2.tgz error (undefined). Will retry in 10 seconds. 2 retries left.
.../share/pnpm/store/v3/tmp/dlx-683229   | Progress: resolved 39, reused 0, downloaded 0, added 0
```

I'm sure `error (undefined)` can't tell me anything. I don't know what went wrong.

**if you use proot-distro in termux to install manjaro, and install pnpm in this way, same error will occur.**

I just followed the steps in [Astro document](https://docs.astro.build/en/install/auto/):
> Run the following command in your terminal to start our handy install wizard:
> `pnpm create astro@latest`
> You can run create astro anywhere on your machine, so there’s no need to create a new empty directory for your project before you begin. If you don’t have an empty directory yet for your new project, the wizard will help create one for you automatically.

I searched for `error (undefined). Will retry in 10 seconds. 2 retries left.`, `pnpm error (undefined)`, but I didn't find anything that may help me.
I don't know much about programming. If you can find any solutions that I didn't see in search results, or I used the wrong keywords, please tell me, I would be appreciated.

## More information about my device

```
[oceancat@raspi4 ~]$ uname -a
Linux raspi4 5.15.92-1-MANJARO-ARM-RPI #1 SMP PREEMPT Mon Feb 6 22:02:34 UTC 2023 aarch64 GNU/Linux
[oceancat@raspi4 ~]$ node -v
v21.1.0
[oceancat@raspi4 ~]$ pnpm -v
8.7.0
```

I know this environment is really weird. But it's one of the only few things I can use.

**Comment by BingXuanOwO** [comment, marked as answer]
This problem seems to because the old version of pnpm. I successfully reproduce this issue on a arm version of manjaro in proot-distro on termux, and pnpm is installed by pacman. Simply uninstall `pnpm` package in pacman and instead to use `corepack enable` to install pnpm and issue just goes away.

**Reply by ideaexplorer233** [reply]
It works. `sudo corepack enable` solved everything. I should really follow the pnpm documentation.
